src/main.f90

```diff
@@ -43,6 +43,9 @@ program main
   integer :: n_wav
   real(dp) :: wav_min, wav_max
 
+
+  integer :: n_threads=1 !needed for omp parallel version
```

Comment:
Just to check, is it not possible to set the number of threads using the OMP_NUM_THREADS environment variable, which would mean no needing to define this variable?

Reply:
In principle yes, but, my experience with the environment variables has been iffy at best, with them not always creating the desired behaviour, while omp_set_num_threads() always works. As a result, I tend to prefer the omp directive. 

However, omp_set_num_threads() will override OMP_NUM_THREADS if it exists, which can be good or bad - a user needs to remember to set two things if they're using more than one package, but they also have the flexibility to easily change behaviour independently. 

In short, it should work either way, but this way is explicit.

Reply:
I wanted to come back to this - it wasn't clear to me from your comment if you wanted me to make a change or just explain the rationale behind the implementation. Given that you haven't resolved the comment I'm assuming you want the change implemented, but it would be good to clarify that before I make more updates if you are interested in merging the PR.